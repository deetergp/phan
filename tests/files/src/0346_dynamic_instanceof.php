<?php

class C346 {}

function f346(C346 $p) {}
function test346() {
    $v1 = 'C346';
    $v2 = new C346;
    if ($v2 instanceof $v1) {
        f346($v2);
        intdiv($v2, 3);  // should warn.
    }
}

function test346B($v1) {
    $v2 = new C346;
    if ($v1 instanceof $v2) {
        f346($v1);
        echo intdiv($v1, 2);  // should warn.
    }
}
function test346C($v1) {
    $v2 = 2;
    if ($v1 instanceof $v2) {  // should warn but should never set the type to int.
        f346($v1);
    }
}
