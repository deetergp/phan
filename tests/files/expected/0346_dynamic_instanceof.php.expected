%s:11 PhanTypeMismatchArgumentInternal Argument 1 (numerator) is \C346 but \intdiv() takes int
%s:19 PhanTypeMismatchArgumentInternal Argument 1 (numerator) is \C346 but \intdiv() takes int
%s:24 PhanTypeInvalidInstanceof Found an instanceof class name of type int, but class name must be a valid object or a string
